### PR TITLE
[WT12] Trivial logger

### PR DESCRIPTION
**Pull request by Sckorn**

### CODE REVIEW

File: common/include/common/logger/trivial_logger.hpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#ifndef RDMN_COMMON_TRIVIAL_LOGGER_HPP
+#define RDMN_COMMON_TRIVIAL_LOGGER_HPP
+
+#include <sstream>
+#include <iostream>
```
Comment (radameon) [inline common/include/common/logger/trivial_logger.hpp:5]: Still needed?

File: common/include/common/logger/trivial_logger.hpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+#include <rdmn-engine/public_api.hpp>
+#include <rdmn-engine/logger/LoggerType.hpp>
+
+namespace rdmn::core::log {
+
+/**
+ * @brief 
+ * Current trivial_logger implementation is created to mimic Boost::Log semantics
+ * in order to reduce th needed changes when doing refactoring of many,
```
Comment (radameon) [inline common/include/common/logger/trivial_logger.hpp:15]: `...the need...`

File: common/include/common/logger/trivial_logger.hpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * and I mean MANY lines of logging code.
+ * It is 
+ * - not efficient
+ * - not robust
+ * - not practical
+ * - and overall a necessary evil
+ * 
+ * This implementation will be replacedby a proper logger, which will be supplied to classes which require it,
```
Comment (radameon) [inline common/include/common/logger/trivial_logger.hpp:23]: `...replaced by...`

File: common/include/common/logger/trivial_logger.hpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    template <size_t Size>
+    R_PUBLIC_API trivial_logger &&operator<<(const char (&cstr)[Size]) && {
+        for (size_t i = 0; i < Size - 1; ++i) {
+            iss_ << cstr[i];
+        }
+
+        if (cstr[Size - 1] != '\0')
+            iss_ << '\0';
```
Comment (radameon) [inline common/include/common/logger/trivial_logger.hpp:58]: This function looks pretty strange, why in the loop above it's `i < Size - 1` and then you have zero terminator check and addtition?

File: common/src/logger/trivial_logger.cpp [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+#elif defined(R_USE_GLOG)
+    LOG(INFO) << str;
+#endif
+}
+
+R_PUBLIC_API void log_debug(std::string &&str) {
+#ifdef R_USE_BOOST_LOG
+    BOOST_LOG_TRIVIAL(debug) << str;
+#elif defined(R_USE_GLOG)
+    LOG(INFO) << str; // no trace in GLOG
```
Comment (radameon) [inline common/src/logger/trivial_logger.cpp:33]: `no debug in GLOG`

File: common/src/logger/trivial_logger.cpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+#elif defined(R_USE_GLOG)
+    LOG(INFO) << str; // no trace in GLOG 
+#endif
+}
+
+R_PUBLIC_API void log_warn(std::string &&str) {
+#ifdef R_USE_BOOST_LOG
+    BOOST_LOG_TRIVIAL(warning) << str;
+#elif defined(R_USE_GLOG)
+    LOG(WARNING) << str; // no trace in GLOG
```
Comment (radameon) [inline common/src/logger/trivial_logger.cpp:41]: Comment is wron, there is a warn log level in GLOG.

File: common/src/logger/trivial_logger.cpp [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+#elif defined(R_USE_GLOG)
+    LOG(ERROR) << str; // no trace in GLOG 
+#endif
+}
+
+R_PUBLIC_API void log_fatal(std::string &&str) {
+#ifdef R_USE_BOOST_LOG
+    BOOST_LOG_TRIVIAL(fatal) << str;
+#elif defined(R_USE_GLOG)
+    LOG(ERROR) << str; // no trace in GLOG
```
Comment (radameon) [inline common/src/logger/trivial_logger.cpp:57]: `// fatal in GLOG terminates the program, so shouldn't be used`

File: common/src/logger/trivial_logger.cpp [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+                log_error(iss_.str());
+            break;
+            case LoggerType::trace:
+                log_trace(iss_.str());
+            break;
+            case LoggerType::debug:
+                log_debug(iss_.str());
+            break;
+            case LoggerType::fatal:
+                log_fatal(iss_.str());
```
Comment (radameon) [inline common/src/logger/trivial_logger.cpp:83]: Mark all of the above `noexcept`

File: thirdparty/glog.BUILD [hunk 9]
```diff
@@ -0,0 +1,3 @@
+load("@com_github_google_glog//:bazel/glog.bzl", "glog_library")
+
+glog_library(0)
```
Comment (radameon) [inline thirdparty/glog.BUILD:3]: Add a comment why.

**Review by radameon — CHANGES_REQUESTED** [review]
Needs some more work

File: common/src/logger/trivial_logger.cpp (hunk 8 above)
Comment (Sckorn) [reply common/src/logger/trivial_logger.cpp:83]: Done

**Review by radameon — CHANGES_REQUESTED** [review]
One more test case is needed.

**Review by radameon — APPROVED** [review]
Squash and merge, if CI is good.

**Pull request merged**